File: skilled_hub_api/test/services/us_address_test.rb
# frozen_string_literal: true

require "test_helper"

class UsAddressTest < ActiveSupport::TestCase
  test "strip_country removes trailing US labels" do
    assert_equal "Houston, TX", UsAddress.strip_country("Houston, TX, United States")
    assert_equal "Memphis, TN", UsAddress.strip_country("Memphis, TN, USA")
    assert_equal "Texas", UsAddress.strip_country("Texas, United States")
  end

  test "parse_city_state splits city and state and ignores country" do
    assert_equal({ city: "Houston", state: "TX" }, UsAddress.parse_city_state("Houston, TX, United States"))
    assert_equal({ city: nil, state: "TX" }, UsAddress.parse_city_state("Texas, United States"))
    assert_equal({ city: "Memphis", state: "TN" }, UsAddress.parse_city_state("Memphis, Tennessee"))
  end
end

File: skilled-hub-frontend/scripts/usAddress.test.mjs
import assert from 'assert';
import { parseCityState, stripCountryFromAddress, usZip5 } from '../src/utils/usAddress.js';

function testStripCountry() {
  assert.strictEqual(stripCountryFromAddress('Houston, TX, United States'), 'Houston, TX');
  assert.strictEqual(stripCountryFromAddress('Memphis, TN, USA'), 'Memphis, TN');
  assert.strictEqual(stripCountryFromAddress('Texas, United States'), 'Texas');
}

function testParseCityState() {
  assert.deepStrictEqual(parseCityState('Houston, TX, United States'), { city: 'Houston', state: 'TX' });
  assert.deepStrictEqual(parseCityState('Texas, United States'), { city: '', state: 'TX' });
  assert.deepStrictEqual(parseCityState('Memphis, Tennessee'), { city: 'Memphis', state: 'TN' });
}

function testZip5() {
  assert.strictEqual(usZip5('77002-1234'), '77002');
  assert.strictEqual(usZip5('abc'), '');
}

testStripCountry();
testParseCityState();
testZip5();
console.log('usAddress tests passed');
